### PR TITLE
feat: Add customisable code themes

### PR DESCRIPTION
**Pull request by liamg**

### CODE REVIEW

File: internal/app/tfsec/cmd/output.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func outputFormat(w io.Writer, addExtension bool, baseFilename, format, fsRoot,
 	switch strings.ToLower(format) {
 	case "lovely", "default":
 		alsoStdout = true
-		factory.WithCustomFormatterFunc(formatter.DefaultWithMetrics(metrics, conciseOutput))
+		factory.WithCustomFormatterFunc(formatter.DefaultWithMetrics(metrics, conciseOutput, codeTheme))
```
Comment (owenrumney) [inline internal/app/tfsec/cmd/output.go:88]: Do we need to validate input here? Or does defsec do it?

File: internal/pkg/formatter/default.go [hunk 2]
```diff
@@ -293,9 +293,22 @@ func printMetadata(w io.Writer, result scan.Result, links []string, isRego bool)
...
+		scan.OptionCodeWithTruncation(true),
+	}
+
+	switch theme {
+	case "dark":
+		codeOpts = append(codeOpts, scan.OptionCodeWithDarkTheme())
+	case "light":
+		codeOpts = append(codeOpts, scan.OptionCodeWithLightTheme())
+	default:
+		codeOpts = append(codeOpts, scan.OptionCodeWithTheme(theme))
```
Comment (owenrumney) [inline internal/pkg/formatter/default.go:308]: ignore previous question

Comment (owenrumney) [reply internal/pkg/formatter/default.go:308]: but what if I write junk in here - does it fallback in defsec?

Comment (liamg) [reply internal/pkg/formatter/default.go:308]: Yeah it falls back to a default dark one.

Comment (liamg) [reply internal/pkg/formatter/default.go:308]: Thought i'd leave it undocumented so we can play with it, but tell people to only use dark/light

**Review by owenrumney — APPROVED** [review]

**Pull request merged**